cmd/controller-manager/app/controller_manager.go

```diff
@@ -279,8 +279,8 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 // getAvailableResources uses the discovery client to determine which API
 // groups are available in the endpoint reachable from the given client and
 // returns a map of them.
-func getAvailableResources(clientBuilder controller.ClientBuilder) (map[schema.GroupVersionResource]bool, error) {
-	var resourceMap []*metav1.APIResourceList
+func getAvailableResources(clientBuilder controller.ClientBuilder, version schema.GroupVersion) (map[schema.GroupVersionResource]struct{}, error) {
```

Comment:
I don't quite follow this. What do we get out of making it a map of void structs instead of bools? Other than that this looks good, but I don't see the reason for this change.

Reply:
@jberkhahn empty struct value is a recommended way of implementing a "set" structure using a map in Go. The difference with boolean is that empty struct doesn't take any space.
See https://www.davidkaya.sk/2017/12/10/sets-in-golang/ for example.

I agree that it's not a critical change, but for me it makes it clear that "this is a set, and value doesn't matter". We never put "false" in this map, so boolean is redundant.